runners/core-java/src/main/java/org/apache/beam/runners/core/metrics/SimpleMonitoringInfoBuilder.java

```diff
@@ -0,0 +1,219 @@
+      List<String> split = Splitter.on(':').splitToList(urn);
+      if (split.size() != 5) {
+        LOG.warn(
+            "Dropping MonitoringInfo for URN {}, UserMetric namespaces and "
+                + "name cannot contain ':' characters.",
+            urn);
+        return false;
+      }
+    } else if (!SimpleMonitoringInfoBuilder.specs.containsKey(urn)) {
+      // Succeed for unknown URNs, this is an extensible metric.
+      return true;
+    } else {
+      spec = SimpleMonitoringInfoBuilder.specs.get(urn);
+    }
+
+    if (!this.builder.getType().equals(spec.getTypeUrn())) {
+      LOG.warn(
+          "Dropping MonitoringInfo since for URN {} with invalid type field. Expected: {}"
+              + " Actual: {}",
+          this.builder.getUrn(),
+          spec.getTypeUrn(),
+          this.builder.getType());
+      return false;
+    }
+
+    Set<String> requiredLabels = new HashSet<String>(spec.getRequiredLabelsList());
+    if (!this.builder.getLabels().keySet().equals(requiredLabels)) {
+      LOG.warn(
+          "Dropping MonitoringInfo since for URN {} with invalid labels. Expected: {}"
+              + " Actual: {}",
+          this.builder.getUrn(),
+          requiredLabels,
+          this.builder.getLabels().keySet());
+      return false;
+    }
+    return true;
+  }
+
+  /** @return The metric URN for a user metric, with a proper URN prefix. */
+  private static String userMetricUrn(String metricNamespace, String metricName) {
+    String fixedMetricNamespace = metricNamespace.replace(':', '_');
+    String fixedMetricName = metricName.replace(':', '_');
+    StringBuilder sb = new StringBuilder();
+    sb.append(USER_COUNTER_URN_PREFIX);
+    sb.append(fixedMetricNamespace);
+    sb.append(':');
+    sb.append(fixedMetricName);
+    return sb.toString();
+  }
+
+  /**
+   * Sets the urn of the MonitoringInfo.
+   *
+   * @param urn The urn of the MonitoringInfo
+   */
+  public void setUrn(String urn) {
+    this.builder.setUrn(urn);
+  }
+
+  /**
+   * Sets the urn of the MonitoringInfo to a proper user metric URN for the given params.
+   *
+   * @param namespace
+   * @param name
+   */
+  public void setUrnForUserMetric(String namespace, String name) {
+    this.builder.setUrn(userMetricUrn(namespace, name));
+  }
+
+  /** Sets the timestamp of the MonitoringInfo to the current time. */
+  public void setTimestampToNow() {
+    Instant time = Instant.now();
+    this.builder.getTimestampBuilder().setSeconds(time.getEpochSecond()).setNanos(time.getNano());
+  }
+
+  /** Sets the int64Value of the CounterData in the MonitoringInfo, and the appropraite type URN. */
+  public void setInt64Value(long value) {
+    this.builder.getMetricBuilder().getCounterDataBuilder().setInt64Value(value);
+    this.builder.setType(SUM_INT64_TYPE_URN);
+  }
+
+  /** Sets the PTRANSFORM MonitoringInfo label to the given param. */
+  public void setPTransformLabel(String pTransform) {
```

Comment:
I think that we can remove setPTransformLabel and setPCollectionLabel and utilize setLabel method inline. We can keep these for convenience as of now.

Reply:
I'd like to keep these here for convenience. Plus, using this is preferred since it will prevent missetting the label key.